lib/impl/refactorRules.js

```diff
@@ -0,0 +1,384 @@
+            }
+
+            //Update the ruleConditionalStatements and targets of the rule.
+            fieldRule.ruleConditionalStatements = formFieldRule.ruleConditionalStatements;
+            fieldRule.targetField = formFieldRule.targetField;
+            fieldRule.save(cb);
+          });
+        }, cb);
+      },
+      function updatePageRulesInDb(cb) {
+        async.each(form.pageRules, function(formPageRule, cb) {
+          var pageRuleId = formPageRule._id.toString();
+          pageRuleModel.findOne({"_id": pageRuleId}, function(err, pageRule) {
+            if (err) {
+              return cb(err);
+            }
+
+            if (pageRule === null) {
+              return cb(new Error("No page rule with id " + pageRuleId + " found."));
+            }
+
+            //Update the ruleConditionalStatements and targets of the rule.
+            pageRule.ruleConditionalStatements = formPageRule.ruleConditionalStatements;
+            pageRule.save(cb);
+          });
+        }, cb);
+      },
+      function deleteFieldRules(cb) {
+        async.eachSeries(fieldRulesToDelete, function(fieldRuleId, cb) {
+          fieldRuleModel.findOne({"_id": fieldRuleId}, function(err, fieldRule) {
+            if (err) {
+              return cb(err);
+            }
+
+            if (fieldRule !== null) {
+              fieldRule.remove(cb);
+            } else {
+              return cb();
+            }
+          });
+        }, cb);
+      },
+      function deletePageRules(cb) {
+        async.eachSeries(pageRulesToDelete, function(pageRuleId, cb) {
+          pageRuleModel.findOne({"_id": pageRuleId}, function(err, pageRule) {
+            if (err) {
+              return cb(err);
+            }
+
+            if (pageRule !== null) {
+              pageRule.remove(cb);
+            } else {
+              return cb();
+            }
+          });
+        }, cb);
+      }
+    ], cb);
+  }
+
+  //Need to delete any rules that are to be deleted
+  function updateFormRules(cb) {
+
+    //Filters out any conditional statements that are no longer valid.
+    function filterConditionalStatements(rule) {
+      return _.filter(rule.ruleConditionalStatements, function(ruleCondStatement) {
+        var sourceField = ruleCondStatement.sourceField.toString();
+
+        if (invalidFields[sourceField]) {
+          /**
+           * If the user flagged the field in this rule for deletion, flag the rule for deletion.
+           */
+          if (ruleDeletionFlags[sourceField]) {
+            rulesFlaggedForDeletion[rule._id.toString()] = true;
+          }
+          return false;
+        } else {
+          return true;
+        }
+      });
+    }
+
+    var updatedFieldRules = _.map(form.fieldRules, function(fieldRule) {
+      var filteredConditionalStatements = filterConditionalStatements(fieldRule);
+      var filteredTargets = _.filter(fieldRule.targetField, function(targetField) {
+        if (invalidFields[targetField]) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1126)